recipes/qemu/meta.yaml

```diff
@@ -0,0 +1,99 @@
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - sysroot_linux-64 2.17  # [linux64]  see below regarding kernel-headers
+    - make
+    - pkg-config
+    - bison
+    - flex
+    - gdb
+    - python # needed to run some tests, not to link against
+  host:
+    - gtk3  # [linux] ?
+    - glib
+    - libfdt
+    - pixman
+      # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+      # MADV_NOHUGEPAGE in a call to madvise.  See:
+      # $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc -iquote $SRC_DIR/build/contrib/libvhost-user -iquote contrib/libvhost-user -iquote $SRC_DIR/tcg/i386 -isystem $SRC_DIR/linux-headers -isystem $SRC_DIR/build/linux-headers -iquote . -iquote $SRC_DIR -iquote $SRC_DIR/accel/tcg -iquote $SRC_DIR/include -iquote $SRC_DIR/disas/libvixl -I$PREFIX/include/pixman-1 -I$SRC_DIR/dtc/libfdt -I$PREFIX/include -pthread -I$PREFIX/include -I$PREFIX/include/glib-2.0 -I$PREFIX/lib/glib-2.0/include -I$PREFIX/include -fPIE -DPIE -m64 -mcx16 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -Wstrict-prototypes -Wredundant-decls -Wall -Wundef -Wwrite-strings -Wmissing-prototypes -fno-strict-aliasing -fno-common -fwrapv -std=gnu99  -Wexpansion-to-defined -Wendif-labels -Wno-shift-negative-value -Wno-missing-include-dirs -Wempty-body -Wnested-externs -Wformat-security -Wformat-y2k -Winit-self -Wignored-qualifiers -Wold-style-declaration -Wold-style-definition -Wtype-limits -fstack-protector-strong -I$PREFIX/include -I$PREFIX/include/libpng16 -I$PREFIX/include -I$SRC_DIR/capstone/include -I$SRC_DIR/tests -I$SRC_DIR/tests/qtest -MMD -MP -MT contrib/libvhost-user/libvhost-user.o -MF contrib/libvhost-user/libvhost-user.d -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -g -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/qemu-5.0.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -c -o contrib/libvhost-user/libvhost-user.o $SRC_DIR/contrib/libvhost-user/libvhost-user.c
+      #/home/conda/staged-recipes/build_artifacts/qemu_1647659211102/work/contrib/libvhost-user/libvhost-user.c: In function 'vu_set_mem_table_exec_postcopy':
+      #/home/conda/staged-recipes/build_artifacts/qemu_1647659211102/work/contrib/libvhost-user/libvhost-user.c:683:23: error: 'MADV_NOHUGEPAGE' undeclared (first use in this function); did you mean 'QEMU_MADV_NOHUGEPAGE'?
+      #  683 |                       MADV_NOHUGEPAGE);
+      #      |                       ^~~~~~~~~~~~~~~
+      #      |                       QEMU_MADV_NOHUGEPAGE
+      # https://man7.org/linux/man-pages/man2/madvise.2.html
+      # I needed the following when building in a non-conda-forge build environment but they give
+      # alternate instructions on using cos7 in https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7 
+
+      #- kernel-headers_linux-64 >=2.6.38  # [linux]
```

Comment:
Look into the kernel-headers being included from `-isystem $SRC_DIR/linux-headers` and configuring it to come from the conda ones instead.  Don't want to be able to build this if it can't be satisfied by the conda sysroot.

Reply:
Actually, that directory contains vendored versions of some of the headers that get updated by https://github.com/qemu/qemu/blob/master/scripts/update-linux-headers.sh.  The comment at the top of that script explains that they are using structs and defines from newer headers.